src/source_util.cpp

```diff
@@ -257,13 +269,33 @@ bool SourceUtil::isPartialMacro(const clang::SourceRange &sourceRange,
 
     const clang::MacroInfo *currentMacro =
         getMacroInfo(end, sourceManager, preprocessor);
+    printMacroTokens(end, sourceManager, preprocessor);
 
     if (sourceManager.isMacroBodyExpansion(end)) {
 
       // Tokens for the current macro's spelling, this only contains the tokens
       // for its definition, included unexpanded identifiers
       llvm::ArrayRef<clang::Token> currentMacroDefTokens =
           currentMacro->tokens();
+
+      // We don't play well with variadic token pasting inside of macros
+      // currently return true
+      if (currentMacro->isVariadic()) {
+        auto hashhashIt = std::find_if(currentMacroDefTokens.begin(),
+                                       currentMacroDefTokens.end(),
+                                       [](const clang::Token &tok) {
+                                         return tok.is(clang::tok::hashhash);
+                                       });
+        // If there is token pasting, ensure that its not followed by variadic
+        // args
+        if (hashhashIt != currentMacroDefTokens.end()) {
```

Comment:
We should probably check this in a loop, right? We could have multiple instances of double hashes, and the one before a `__VA_ARGS__` might not be the first one.
```
#define test(a, ...) f(a ## .value(), ##__VA_ARGS__)
```